pkg/repository/prisma/workflow_run.go

```diff
@@ -725,354 +753,432 @@ func workflowRunMetricsCount(ctx context.Context, pool *pgxpool.Pool, queries *d
+
+			}
 
-				if dedupeInt, ok := dedupeValue.(int); ok {
-					dedupeStr := fmt.Sprintf("%d", dedupeInt)
-					opts.DedupeValue = &dedupeStr
+			if opt.Priority != nil {
+				createParams.Priority = pgtype.Int4{
+					Int32: *opt.Priority,
+					Valid: true,
 				}
 			}
-		}
 
-		if opts.Priority != nil {
-			createParams.Priority = pgtype.Int4{
-				Int32: *opts.Priority,
-				Valid: true,
+			crp := dbsqlc.CreateWorkflowRunsParams{
+				ID:                 createParams.ID,
+				TenantId:           createParams.Tenantid,
+				WorkflowVersionId:  createParams.Workflowversionid,
+				DisplayName:        createParams.DisplayName,
+				ChildIndex:         createParams.ChildIndex,
+				ChildKey:           createParams.ChildKey,
+				ParentId:           createParams.ParentId,
+				ParentStepRunId:    createParams.ParentStepRunId,
+				AdditionalMetadata: createParams.Additionalmetadata,
+				Priority:           createParams.Priority,
+				Status:             "PENDING",
 			}
-		}
 
-		// create the dedupe value
-		if opts.DedupeValue != nil {
-			_, err = queries.CreateWorkflowRunDedupe(
-				tx1Ctx,
-				tx,
-				dbsqlc.CreateWorkflowRunDedupeParams{
-					Tenantid:          pgTenantId,
-					Workflowversionid: sqlchelpers.UUIDFromStr(opts.WorkflowVersionId),
-					Value:             sqlchelpers.TextFromStr(*opts.DedupeValue),
-					Workflowrunid:     sqlchelpers.UUIDFromStr(workflowRunId),
-				},
+			createRunsParams = append(createRunsParams, crp)
+
+			var desiredWorkerId *string
+
+			if opt.DesiredWorkerId != nil {
+				// problem here is that we need a desirewdWorkerId for the sticky state
+				// so we can only create an entry in WorkflowRunStickyState if we have a desiredWorkerId
+
+				desiredWorkerId = opt.DesiredWorkerId
+				stickyInfos = append(stickyInfos, stickyInfo{
+					workflowRunId:     sqlchelpers.UUIDFromStr(workflowRunId),
+					workflowVersionId: sqlchelpers.UUIDFromStr(opt.WorkflowVersionId),
+					desiredWorkerId:   sqlchelpers.UUIDFromStr(*desiredWorkerId),
+				})
+
+			}
+
+			var (
+				eventId, cronParentId, scheduledWorkflowId pgtype.UUID
+				cronSchedule                               pgtype.Text
 			)
 
-			if err != nil {
-				// if this is a unique violation, return stable error
-				if isUniqueViolationOnDedupe(err) {
-					return nil, repository.ErrDedupeValueExists{
-						DedupeValue: *opts.DedupeValue,
-					}
+			if opt.TriggeringEventId != nil {
+				eventId = sqlchelpers.UUIDFromStr(*opt.TriggeringEventId)
+			}
+
+			if opt.CronParentId != nil {
+				cronParentId = sqlchelpers.UUIDFromStr(*opt.CronParentId)
+
+			}
+			if opt.Cron != nil {
+				cronSchedule = sqlchelpers.TextFromStr(*opt.Cron)
+			}
+
+			if opt.ScheduledWorkflowId != nil {
+				scheduledWorkflowId = sqlchelpers.UUIDFromStr(*opt.ScheduledWorkflowId)
+			}
+
+			// so I can't pass a NULL in for parentID so I need to set it to something
+			// setting it to workflowRunId for now but maybe thats not correct
```

Comment:
Ah yeah, I think we got confused yesterday, and the parent id should just be the existing workflowRunId (too many fields named `parentId` :)). The point of this is to uniquely identify the relation of the WorkflowRunTriggeredBy, it doesn't refer to the parent workflow, which is confusing as hell. So just set it to the generated UUID of the workflow run!